plugins/haconfig/keepalived/helpers.go

```diff
@@ -0,0 +1,78 @@
+package keepalived
+
+import (
+	"fmt"
+	"strings"
+
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+type ReporterFunction func(string, error) 
+
+
+//  Delete matching replication controllers.
+func deleteReplicationControllers(name string, namespace string, selector map[string]string, kClient kclient.Interface, after ReporterFunction) {
+	labels := labels.SelectorFromSet(selector)
```

Comment:
We generally just don't do that from this code base - that's why we have stop and delete and others (other ways to do this that are consistent).

----- Original Message -----
> > +
> > +import (
> > +	"fmt"
> > +	"strings"
> > +
> > +	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +)
> > +
> > +type ReporterFunction func(string, error)
> > +
> > +
> > +//  Delete matching replication controllers.
> > +func deleteReplicationControllers(name string, namespace string, selector
> > map[string]string, kClient kclient.Interface, after ReporterFunction) {
> > +	labels := labels.SelectorFromSet(selector)
> 
> So I had a conversation with @abhgupta yesterday and was asking re: the
> cleanup/deletion of these objects and how to go about doing it. The
> ```--delete``` option cleans up the deployment artifacts and uses this.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1807/files#r28802537